src/v/cluster/rm_stm.h

```diff
@@ -395,6 +395,24 @@ class rm_stm final : public persisted_stm {
     struct seq_entry_wrapper {
         seq_entry entry;
         safe_intrusive_list_hook _hook;
+
+        seq_entry_wrapper() = default;
+        explicit seq_entry_wrapper(seq_entry&& entry)
+          : entry(std::move(entry)) {}
+        seq_entry_wrapper(const seq_entry_wrapper&) = delete;
+        seq_entry_wrapper& operator=(seq_entry_wrapper&) = delete;
+        ~seq_entry_wrapper() = default;
+
+        seq_entry_wrapper(seq_entry_wrapper&& other) noexcept
+          : entry(std::move(other.entry)) {
+            _hook.swap_nodes(other._hook);
+        }
+
+        seq_entry_wrapper& operator=(seq_entry_wrapper&& other) noexcept {
+            entry = std::move(other.entry);
+            _hook.swap_nodes(other._hook);
```

Comment:
something that i don't quite understand here: why use swap on the nodes for move assignment? like when i assign the entry i'd expect the value to change but not the list memberships?